osu.Game/Graphics/UserInterface/StarCounter.cs

```diff
@@ -17,29 +17,52 @@
 
 namespace osu.Game.Graphics.UserInterface

```

Comment:
this whole class is really hard to read and structured badly. i'd rewrite it without the weird transform stuff you're doing.